sdk/core/Azure.Core/tests/AzureNamedKeyCredentialTests.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests
+{
+    [TestFixture]
+    public class AzureNamedKeyCredentialTests
+    {
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void ConstructorValidatesTheName(string name)
+        {
+            var exceptionType = name switch
+            {
+                null => typeof(ArgumentNullException),
+                _ => typeof(ArgumentException)
+            };
+
+            Assert.Throws(exceptionType, () => new AzureNamedKeyCredential(name, "key"));
+        }
+
+        [Test]
+        [TestCase(null)]
+        [TestCase("")]
+        public void ConstructorValidatesTheKey(string key)
+        {
+            var exceptionType = key switch
```

Comment:
nit: If you pass the expected exception type in as an additional argument, you can avoid the switch and the test body becomes just the assert.

Reply:
ohhh.... I didn't think of that!   I like it.